sparse/rest/handlers.go

```diff
@@ -0,0 +1,226 @@
+package rest
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"sort"
+	"strconv"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/rancher/sparse-tools/sparse"
+)
+
+func (server *SyncServer) sendErrorResponse(errorMessage string, writer http.ResponseWriter) {
+	log.Error(errorMessage)
+	http.Error(writer, errorMessage, http.StatusInternalServerError)
+}
+
+func (server *SyncServer) getQueryInterval(request *http.Request) (sparse.Interval, error) {
+	var interval sparse.Interval
+	var err error
+
+	queryParams := request.URL.Query()
+	if len(queryParams["begin"]) == 0 {
+		err = fmt.Errorf("queryParams begin not exist")
+		return interval, err
+	}
+	if len(queryParams["end"]) == 0 {
+		err = fmt.Errorf("queryParams end not exist")
+		return interval, err
+	}
+
+	beginStr := queryParams["begin"][0]
+	endStr := queryParams["end"][0]
+	log.Debugf("file offset begin: %s, end: %s", beginStr, endStr)
+
+	begin, err := strconv.ParseInt(beginStr, 10, 64)
+	if err != nil {
+		err = fmt.Errorf("strconv.ParseInt(beginStr) error: %s", err)
+		return interval, err
+	}
+	end, err := strconv.ParseInt(endStr, 10, 64)
+	if err != nil {
+		err = fmt.Errorf("strconv.ParseInt(endStr) error: %s", err)
+		return interval, err
+	}
+
+	return sparse.Interval{begin, end}, err
+}
+
+func (server *SyncServer) open(writer http.ResponseWriter, request *http.Request) {
+	log.Debug("open called")
+
+	// get file size
+	interval, err := server.getQueryInterval(request)
+	if err != nil {
+		errorMessage := fmt.Sprintf("server.getQueryInterval failed, err: %s", err)
+		server.sendErrorResponse(errorMessage, writer)
+		return
+	}
+
+	// get filePath
+	queryParams := request.URL.Query()
+	if len(queryParams["filePath"]) == 0 {
+		errorMessage := fmt.Sprintf("server.open failed, queryParams filePath not exist")
+		server.sendErrorResponse(errorMessage, writer)
+		return
+	}
+	filePath := queryParams["filePath"][0]
```

Comment:
Get rid of filepath here, should be passed in through CLI.

As @ibuildthecloud pointed out, there is also security risk for server to write into any file.